core/errors/infer.h

```diff
@@ -37,7 +37,7 @@ constexpr ErrorClass UntypedConstantSuggestion{7027, StrictLevel::Strict};
 // constexpr ErrorClass LazyResolve{7029, StrictLevel::True};
 constexpr ErrorClass MetaTypeDispatchCall{7030, StrictLevel::True};
 constexpr ErrorClass PrivateMethod{7031, StrictLevel::True};
-constexpr ErrorClass GenericArgumentKeywordArgs{7032, StrictLevel::True};
+// constexpr ErrorClass GenericArgumentKeywordArgs{7032, StrictLevel::True};
```

Comment:
I think we probably want to keep the infer error as well.

In `# typed: true` files, Sorbet actually reports the error twice, and having two different error codes makes it a little easier to see why that is.

Also, we need to report this error even if the type annotation does not occur in a signature—i.e., even if it occurs in a value position:

[example](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aextend%20T%3A%3ASig%0A%0Aclass%20Box%0A%20%20extend%20T%3A%3AGeneric%0A%20%20Elem%20%3D%20type_member%0Aend%0A%0Asig%20%7Breturns%28Box%5BInteger%2C%20x%3A%20Integer%5D%29%7D%0A%23%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%20error%3A%20Keyword%20arguments%20given%20to%20%60Array%60%0Adef%20example%0A%20%20raise%0Aend%0A)

In that case, it would be weird for the call to `Box[...].new` to have an error code starting with 5xxx when the error actually comes from infer.

This is kind of just a weird thing about how Sorbet's choice to implement type syntax works (in an alternative language, you'd just have the parser figure our which positions are type syntax and which are expression syntax, and then only need to report these errors once) but in Sorbet since type annotations are also expressions, sometimes the errors double up.

Reply:
Got it, that example makes it a lot clearer now. I hadn't thought about how the dual use of type syntax could cause overlaps, but it totally makes sense why keeping the infer error would help. Interesting quirk compared to something like TypeScript, for sure. I'll go ahead and revert the error classes to infer, too.